### DISCUSSION TITLE
Custom command-line tool EZDXF

### DISCUSSION THREAD

**Discussion by realhoolito** [mozman/ezdxf#1104] [Q&A]
We just come across this online tool but we cannot use it because we don't know anything about Python. We have a strange problem exporting R12 DXFs from the latest version of Siemens NX CAD, which have errors being imported in iMill, an ancient CAM software but unfortunately our only gateway to drive our CNC mill. iMill only seems to accept R12 DXFs from Draftsight, for some strange reason we are almost sure someone here of the developers of EZDXF could discover by comparing both DXFs

We are hoping a pass through EZDXF could solve this. Can some one here, help?

Thanks! Regards,

Disclaimer: I just know python is a snake.
We wouldn't know how to run EZDXF

**Comment by mozman** [comment]
HI!

The first and easiest way is to run the `audit` command on your DXF files.

I guess you are on Windows.

1. Install the latest [Python](https://www.python.org/downloads/) version, make sure you install the Python launcher `py` (installed by default)
2. install `ezdxf`:

    py -m pip install ezdxf

3. run the audit command:

    py -m ezdxf audit -s <your.dxf>

The `-s` flag saves the recovered file as <your.rec.dxf>

**Reply by realhoolito** [reply]
Hi. thanks a lot for this first step!
On the file that I know _doesn't_ work*, audit says no errors. But the .rec file does show tiny differences.
For example a number that was -0.5, gets saved as 1e+20 in the .rec file
Tomorrow I'll check if that file gets imported differently in the CAM software
Tomorrow I'll also try to get the DXF run through DraftSight and audit that file as well. 

I learned in the documentation I can use "ezdxf browse", but it doesn't work in this installation yet: I get "No Qt binding, tried PySide6 and PyQt5"
It might be interesting seeing the file that way

* Doesn't work = DXF loads in CAM, but some closed loops are reported as open. Although I can't see anything weird in the source CAD.

**Reply by mozman** [reply]
Just add PySide6 to your installation:

    py -m pip install pyside6

or update `ezdxf` with all drawing features:

    py -m pip install -U ezdxf[draw]

**Reply by realhoolito** [reply]
Thanks a lot for this!
We suspect a distance threshold or tolerance issue. 
NX export doesn't have the option to merge points less than "x" distance, whereas Solidworks does. Solidworks do work even after merely importing-exporting NX files. 
I'll pick this back up if problems arise again.

Your help was valuable, have a nice week!
